### PR TITLE
Reducers and actions

### PR DESCRIPTION
**Pull request by MohamedHNoor**
# Added reducers and actions using:

- [x]  store, actions, and reducers in React
- [x] Redux Store and create actions and reducer for adding and removing books
- [x] Structure file application using a "feature folder" approach

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      state.push(action.payload);
+    },
+    removeBook: (state, action) =>
+      // eslint-disable-next-line implicit-arrow-linebreak
+      state.filter((book) => book.id !== action.payload),
+  },
+});
+
+export const { addBook, removeBook } = booksSlice.actions;
+
```
Comment (omar-muhamad) [inline src/redux/books/books.js]: Great job using `createSlice` for writing your reducer as suggested by `redux-toolkit`, but please consider the following required changes:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md#:~:text=Define%20action%20types%20for%20adding%20and%20removing%20a%20book.), Please define action types for adding and removing a book using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
- [ ] Kindly delete the `eslint` disablement line 11 and fix it yourself.

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState: [], // initial state is an empty array
+  reducers: {
+    // reducers are functions that determine changes to the state
+    checkCategory: () => 'Under construction',
+  },
+});
+
+export const { checkCategory } = categoriesSlice.actions;
```
Comment (omar-muhamad) [inline src/redux/categories/categories.js]: - [ ] Same here, As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md#:~:text=Define%20an%20action%20type%20for%20checking%20the%20status.), Please define an action type for checking the status using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @MohamedHNoor,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @MohamedHNoor,

✔️ Implementation of ducks pattern for action types

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**